Updated for clarity and for YAML spacing (#6734)
harness/developer-hub: docs/platform/delegates/secure-delegates/delegate-mtls-support.md

   You provide the `client.crt` and `client.key` to the delegate YAML when you install the delegate.
   :::

5. After you create the certificates, provide the public cert of the CA certificate and a desired unique subdomain to Harness support.

   :::info note
   After this, Harness will perform the steps to enable the mTLS.
   :::

## Enable mTLS on delegate

> You can enabled mTLS on delegates with the image tag in `yy.mm.xxxxx` format. Legacy delegates are not supported. For more information on delegate types, go to [Delegate image types](/docs/platform/delegates/delegate-concepts/delegate-image-types).

Reviewer: ```suggestion
You can enable mTLS on delegates with the image tag in `yy.mm.xxxxx` format. Legacy delegates are not supported. For more information on delegate types, go to [Delegate image types](/docs/platform/delegates/delegate-concepts/delegate-image-types).
```

### Prerequisites for enabling mTLS on a delegate

Before you enable mTLS on a delegate, make sure that you meet the following prerequisites.

- The delegate must have an image tag in _`yy.mm.xxxxx`_ format. It cannot be a legacy delegate.

- You must have access to the delegate YAML.

### Configure mTLS on delegate

<Tabs>
  <TabItem value="k8s" label="Kubernetes delegate" delegate>

  To enable mTLS on a Kubernetes delegate, you have to create a secret and update the delegate YAML. For an example Kubernetes manifest, go to [Sample Kubernetes manifest](https://github.com/harness/delegate-kubernetes-manifest/blob/main/harness-delegate.yaml).

>   1. From the same folder where you have your client.crt and client.key files, run the following command to create the secret

Reviewer: ```suggestion
  1. From the same folder where you have your `client.crt` and `client.key` files, run the following command to create the secret.
```

                      defaultMode: 400
          ```      

  3. In the `ConfigMap` resource of manifest YAML file, make the following updates:

      1. Under `data.config.yaml`, update the value for `managerHost` to `https://<subdomain>.agent.harness.io`.

      2. Under `data.config.yaml`, add the following YAML.

          ```yaml
              clientCertificateFilePath: /etc/mtls/client.crt
              clientCertificateKeyFilePath: /etc/mtls/client.key
          ```   

  4. In the `CronJob` resource of manifest YAML file, make the following updates:

      1. Under `jobTemplate.spec.template.spec.containers.volumeMounts`, add the following YAML.
    
          ```yaml
                        - name: client-certificate
                          mountPath: /etc/mtls
                          readOnly: true
          ```   

      2. Under `jobTemplate.spec.template.spec.volumes`, add the following YAML.
      
          ```yaml
                      - name: client-certificate
                        secret:
                          secretName: client-certificate
          ```      

  5. Save and apply the manifest.

  </TabItem>

>   <TabItem value="docker" label="Docker delegate">

Reviewer: ```suggestion
  <TabItem value="docker" label="Docker delegate">
  
```

>     To enable mTLS on a Docker delegate, perform the following commands.

Reviewer: ```suggestion
    To enable mTLS on a Docker delegate, do the following:
```

    1. Copy the following example command.

        ```
        docker run -d --cpus=1 --memory=2g -u root -v <path to client certificate>:/etc/mtls/client.crt -v <path to client key>:/etc/mtls/client.key \
          -e DELEGATE_NAME=docker-delegate \
          -e NEXT_GEN="true" \
          -e DELEGATE_TYPE="DOCKER" \
          -e ACCOUNT_ID=<account ID> \
          -e DELEGATE_CLIENT_CERTIFICATE_PATH=/etc/mtls/client.crt \
          -e DELEGATE_CLIENT_CERTIFICATE_KEY_PATH=/etc/mtls/client.key \
          -e DELEGATE_TOKEN=<delegate token> \
          -e LOG_STREAMING_SERVICE_URL=https://<subdomain>.agent.harness.io/log-service/ \
          -e MANAGER_HOST_AND_PORT=https://<subdomain>.agent.harness.io harness/delegate:yy.mm.verno
        ```

>     2. Update all of the placeholders: `<path to client certificate>`, `<path to client key>`, `<account ID>`, `<delegate token>`, and `<subdomain>`

Reviewer: ```suggestion
    2. Update all of the placeholders: `<path to client certificate>`, `<path to client key>`, `<account ID>`, `<delegate token>`, and `<subdomain>`.
```
